### ISSUE TITLE
Nerdtree window switches/swaps with main window,  when opened with directory as argument

### ISSUE THREAD

**Issue by DevT0ny** [preservim/nerdtree#1272]
#### Environment
- Operating System: x86_64 GNU/Linux, 5.4.45
- Vim/Neovim version `:version`: v0.5.1
- NERDTree version, found on first line of quickhelp `?`: 6.10.16
- Are you using any of these NERDTree-dependent plugins?  
    - [ ] [Xuyuanp/nerdtree-git-plugin](https://github.com/Xuyuanp/nerdtree-git-plugin)
    - [ ] [ryanoasis/vim-devicons](https://github.com/ryanoasis/vim-devicons)
    - [ ] [tiagofumo/vim-nerdtree-syntax-highlight](https://github.com/tiagofumo/vim-nerdtree-syntax-highlight)
    - [ ] [scrooloose/nerdtree-project-plugin](https://github.com/scrooloose/nerdtree-project-plugin)
    - [ ] [PhilRunninger/nerdtree-buffer-ops](https://github.com/PhilRunninger/nerdtree-buffer-ops)
    - [ ] [PhilRunninger/nerdtree-visual-selection](https://github.com/PhilRunninger/nerdtree-visual-selection)
    - [ ] [jistr/vim-nerdtree-tabs](https://github.com/jistr/vim-nerdtree-tabs)
    - [ ] Others (specify):
- Provide a minimal **.vimrc** file that will reproduce the issue.
```vim
if has("nvim")
  let g:plug_home = stdpath('data') . '/plugged'
endif

call plug#begin()
Plug 'preservim/nerdtree'
call plug#end()

autocmd!  
set number relativenumber
syntax enable
set encoding=utf-8
set background=dark
set hidden 
set confirm
set nobackup
set nowritebackup
set noswapfile
set expandtab
set shiftwidth=2
set softtabstop=2
set tabstop=2
set termguicolors

let mapleader = "\<Space>"
nnoremap <silent> Q :q<CR>
nnoremap <silent><C-n> :NERDTreeToggle<CR>

let g:NERDTreeChDirMode=2
let g:NERDTreeIgnore=['\.rbc$', '\~$', '\.pyc$', '\.db$', '\.sqlite$', '^__pycache__$', '^node_modules$']
let g:NERDTreeSortOrder=['^__\.py$', '\/$', '*', '\.swp$', '\.bak$', '\~$']
let g:NERDTreeShowBookmarks=1
let g:nerdtree_tabs_focus_on_files=1
let g:NERDTreeGitStatusWithFlags = 1

" How can I close vim if the only window left open is a NERDTree?
"autocmd bufenter * if (winnr("$") == 1 && exists("b:NERDTree") && b:NERDTree.isTabTree()) | q | endif

" If another buffer tries to replace NERDTree, put it in the other window, and bring back NERDTree.
autocmd BufEnter * if bufname('#') =~ 'NERD_tree_\d\+' && bufname('%') !~ 'NERD_tree_\d\+' && winnr('$') > 1 |
    \ let buf=bufnr() | buffer# | execute "normal! \<C-W>w" | execute 'buffer'.buf | endif

" Open the existing NERDTree on each new tab.
"autocmd BufWinEnter * if getcmdwintype() == '' | silent NERDTreeMirror | endif

```

#### Steps to Reproduce the Issue
1. Invoke nvim with directory as argument
2. navigate and try to open any file.
3. invoke nerdtree (`:NERDTreeToggle`)
4. try to switch between nerdtree window and file window. you can use any  map`<c-w><c-h>`,`<c-w>w` or `:wincmd h`.

https://user-images.githubusercontent.com/47944176/135973428-685b5bdf-e284-4abd-8e47-4f637d43655a.mp4

**Comment by astier** [comment]
Are you sure this is a minimal vimrc?

**Comment by acdifran** [comment]
@DevT0ny I actually ran into something similar recently and didn't want to remove that snippet since it is helpful to not replace NERDTree with other buffers. When a new buffer is opened in a new window, bufname('#') will always be different from bufname('%') at first, so this is triggering the buffer swap.

I modified it to only do the buffer swap if the window where BufEnter is called is the leftmost window since my NERDTree should always be the leftmost window. This way, it still works when that window is replaced with a  different buffer, but doesn't do the swap when you move from the NERDTree window to another window.

```vim script
autocmd BufEnter * if winnr() == winnr('h') && bufname('#') =~ 'NERD_tree_\d\+' && bufname('%') !~ 'NERD_tree_\d\+' && winnr('$') > 1 |
    \ let buf=bufnr() | buffer# | execute "normal! \<C-W>w" | execute 'buffer'.buf | endif
```
